File: 06_Algorithm/leetcode_589.go
/**
 * File              : leetcode_589.go
 * Author            : Andy
 * Date              : 2022.08.20
 * Last Modified Date: 2022.08.20
 * Last Modified By  : Andy
 */

package preorder

func preorder(root *Node) []int {
	ans := make([]int, 0)
	finder(root, &ans)
	return ans
}

func finder(root *Node, ans *[]int) {
	if root == nil {
		return
	}
	for _, node := range root.Children {
		finder(node, ans)
	}
}

File: Algorithm/leetcode_589.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_589.py
# Author            : Andy
# Date              : 2022.08.20
# Last Modified Date: 2022.08.20
# Last Modified By  : Andy


class Solution:
    def preorder(self, root):
        def find(root):
            if root is None:return
            ans.append(root.val)
            for child in root.Children:
                find(child)
        ans = []
        rind(root)
        return ans
